src/server/pfs/server/driver.go

```diff
@@ -791,17 +787,14 @@ func (d *driver) propagateBranches(txnCtx *txncontext.TransactionContext, branch
 			if err != nil {
 				return err
 			}
-			if provOfSubvBI.Head == nil || provOfSubvBI.Branch.Repo.Type == pfs.SpecRepoType {
```

Comment:
What was the reason for filtering out the spec repo originally?

Reply:
The idea is that it doesn't contribute datums, so we would only create an output commit (and thus a job) based on input commits.  Currently in global IDs, we will create an initial empty job (unless the pipeline has input datums on creation).

Reply:
Wouldn't you need a new output commit for the reprocessing case?